jni/lib/src/accessors.dart

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:ffi';
+import 'package:ffi/ffi.dart' show using;
+
+import 'package:jni/src/jvalues.dart';
+
+import 'third_party/jni_bindings_generated.dart';
+import 'jni.dart';
+import 'env_extensions.dart';
+
+Pointer<GlobalJniEnv> env = Jni.env;
+
+void _check(JThrowable exception) {
+  if (exception != nullptr) {
+    env.throwException(exception);
+  }
+}
+
+extension JniResultMethods on JniResult {
+  void check() => _check(exception);
+
+  int get byte {
```

Comment:
Nice! Neat way to throw exceptions.

Also a decent way to get "methods" on generated ffi structs. :)